core/src/main/java/com/graphhopper/routing/ch/CHProfileSelector.java

```diff
@@ -65,10 +62,7 @@ private CHProfile select() {
             if (uTurnCosts != null && p.getUTurnCostsInt() != uTurnCosts) {
                 continue;
             }
-            if (!weighting.isEmpty() && !getWeighting(p).equals(weighting)) {
-                continue;
-            }
-            if (!vehicle.isEmpty() && !getVehicle(p).equals(vehicle)) {
+            if (!p.getWeighting().matches(hintsMap)) {
```

Comment:
If we do this we probably should use `.matches()` also a few lines below (lines 80+81)

Reply:
In lines 80+81 we compare to Weightings, but here we compare the matching of a Weighting to the HintsMap. So, to use the matches method we would need `getWeighting().matches(hintsMap) && getWeighting().matches(hintsMap)` - and still both Weighting objects could be different. Or to remove the vehicle comparison we could use the Weighting.equals method instead of String.equals?

Reply:
Maybe we can get rid of `Weighting#matches` altogether :)

Reply:
Yeah, have created a separate issue as this seems to be quite involved: https://github.com/graphhopper/graphhopper/issues/1800

Will merge this here for now if no objections :) ?